### PR TITLE
Add title, description, code, and source_location to example metadata

### PR DESCRIPTION
**Pull request by vjeffrey**
screen shot 2016-03-17 at 9 40 22 am

### CODE REVIEW

File: lib/inspec/rspec_json_formatter.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@ def format_example(example)
         pending_message: example.execution_result.pending_message,
         id: example.metadata[:id],
         impact: example.metadata[:impact],
+        title: example.metadata[:title],
+        descr: example.metadata[:desc],
```
Comment (arlimus) [inline lib/inspec/rspec_json_formatter.rb]: Could we use `:desc` instead of `:descr`?

**Comment by chris-rock** [comment]
@vjeffrey +1 I like that implementation. Could we cover the `title`, `desc` in a separate object:

```
{
    ...
   metadata : {
      title: "abc",
      desc: ""
  }
}
```

**Comment by arlimus** [comment]
Great point @chris-rock , let's move to a clean separate JSON format that combines inspec's fields and rspec's data to a proper report structure.

**Comment by arlimus** [comment]
For now: flatten it out until we have another json reporter. Please don't use these fields here, they will go away!

**Comment by arlimus** [comment]
Please not that this has been superseded by https://github.com/chef/inspec/pull/552

**Pull request merged**